backend/python/app/services/entity_service.py

```diff
@@ -0,0 +1,46 @@
+from ..models.entity import Entity
+from ..models import db
+
+
+def get_entities():
+    # Entity is a SQLAlchemy model, we can use convenient methods provided
+    # by SQLAlchemy like query.all() to query the data
+    return [result.to_dict() for result in Entity.query.all()]
+
+
+def get_entity(id):
+    # get queries by the primary key, which is id for the Entity table
+    entity = Entity.query.get(id)
+    if entity is None:
+        raise ValueError
+    return entity.to_dict()
+
+
+def create_entity(entity):
+    new_entity = Entity(**entity)
+    db.session.add(new_entity)
```

Comment:
Is it possible for this to raise an exception (what happens if validation on a field fails)? If so the error case should be handled

Reply:
We already check this in routes, so I believe it is not possible for it to raise an exception.